nifi-registry/nifi-registry-core/nifi-registry-jetty/src/main/java/org/apache/nifi/registry/jetty/JettyServer.java

```diff
@@ -178,303 +118,60 @@ private void configureConnectors() {
         try {
-            // start the server
             server.start();
 
-            // ensure everything started successfully
-            for (Handler handler : server.getChildHandlers()) {
-                // see if the handler is a web app
-                if (handler instanceof WebAppContext) {
-                    WebAppContext context = (WebAppContext) handler;
+            final Optional<Throwable> unavailableExceptionFound = Arrays.stream(server.getChildHandlers())
+                    .filter(handler -> handler instanceof WebAppContext)
+                    .map(handler -> (WebAppContext) handler)
+                    .map(WebAppContext::getUnavailableException)
+                    .filter(Objects::nonNull)
+                    .findFirst();
 
-                    // see if this webapp had any exceptions that would
-                    // cause it to be unavailable
-                    if (context.getUnavailableException() != null) {
-                        startUpFailure(context.getUnavailableException());
-                    }
-                }
+            if (unavailableExceptionFound.isPresent()) {
+                final Throwable unavailableException = unavailableExceptionFound.get();
+                shutdown(unavailableException);
             }
 
-            dumpUrls();
-        } catch (final Throwable t) {
-            startUpFailure(t);
-        }
-    }
-
-    private void startUpFailure(Throwable t) {
-        System.err.println("Failed to start web server: " + t.getMessage());
-        System.err.println("Shutting down...");
-        logger.warn("Failed to start web server... shutting down.", t);
-        System.exit(1);
-    }
-
-    private void dumpUrls() throws SocketException {
-        final List<String> urls = new ArrayList<>();
-
-        for (Connector connector : server.getConnectors()) {
-            if (connector instanceof ServerConnector) {
-                final ServerConnector serverConnector = (ServerConnector) connector;
-
-                Set<String> hosts = new HashSet<>();
-
-                // determine the hosts
-                if (StringUtils.isNotBlank(serverConnector.getHost())) {
-                    hosts.add(serverConnector.getHost());
-                } else {
-                    Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
-                    if (networkInterfaces != null) {
-                        for (NetworkInterface networkInterface : Collections.list(networkInterfaces)) {
-                            for (InetAddress inetAddress : Collections.list(networkInterface.getInetAddresses())) {
-                                hosts.add(inetAddress.getHostAddress());
-                            }
-                        }
-                    }
+            final List<URI> applicationUrls = getApplicationUrls();
+            if (applicationUrls.isEmpty()) {
+                logger.warn("Started Server without connectors");
```

Comment:
If this PR is applied to `support/nifi-1.x`, we should be aware that the server startup text is changing.  Prior to this PR, the startup text looked like this:
```
2023-05-22 06:14:36,044 INFO [main] org.eclipse.jetty.server.Server Started @10783ms
2023-05-22 06:14:36,048 INFO [main] o.apache.nifi.registry.jetty.JettyServer NiFi Registry has started. The UI is available at the following URLs:
2023-05-22 06:14:36,049 INFO [main] o.apache.nifi.registry.jetty.JettyServer http://192.168.1.146:18080/nifi-registry
2023-05-22 06:14:36,049 INFO [main] o.apache.nifi.registry.jetty.JettyServer http://10.19.8.251:18080/nifi-registry
2023-05-22 06:14:36,049 INFO [main] o.apache.nifi.registry.jetty.JettyServer http://127.0.0.1:18080/nifi-registry
2023-05-22 06:14:36,050 INFO [main] o.apache.nifi.registry.BootstrapListener Successfully initiated communication with Bootstrap
2023-05-22 06:14:36,050 INFO [main] org.apache.nifi.registry.NiFiRegistry Registry initialization took 9841683488 nanoseconds (9 seconds).
```

Now it looks like this:
```
2023-05-22 06:23:40,314 INFO [main] org.eclipse.jetty.server.Server Started @10053ms
2023-05-22 06:23:40,315 INFO [main] o.apache.nifi.registry.jetty.JettyServer Started Server on http://0.0.0.0:18080/nifi-registry
2023-05-22 06:23:40,316 INFO [main] o.apache.nifi.registry.BootstrapListener Successfully initiated communication with Bootstrap
2023-05-22 06:23:40,317 INFO [main] org.apache.nifi.registry.NiFiRegistry Started Application in 9.26 seconds (9260276707 ns)
```

This may not be an issue, but I wanted to at least point it out in case there are any users who detect server startup using any of these log statements.  The line `org.eclipse.jetty.server.Server Started` is at least consistent.

Reply:
Thanks for highlighting this change @gresockj.

The adjustment to the listening addresses is intentional as it reflects the Jetty Server Connector address of `0.0.0.0` when not explicitly configured. The current approach enumerates all available interfaces and addresses, which is correct in the sense that `0.0.0.0` means all interfaces, but the change brings the NiFi Registry logs in line with Jetty configuration.

The startup time logging change reflects more recent adjustments in NiFi where the rounded number of seconds is listed, followed by the nanonseconds.

I think the change should be backported, but thanks for highlighting the differences.